plutus-tx-plugin/src/PlutusTx/Compiler/Builtins.hs

```diff
@@ -157,6 +157,7 @@ mkBuiltin = PIR.Builtin ()
 builtinNames :: [TH.Name]
 builtinNames = [
       ''BS.ByteString
```

Comment:
can remove this now, I think

Reply:
The compilation fails without it:

```
Exception: Error: Unexpected error during compilation, please report this to the Plutus team: Missing builtin name: Data.ByteString.Internal.ByteString
```

And with changing `defineBuiltinType ''BS.ByteString $ PLC.toTypeAst $ Proxy @BS.ByteString`:

```
    traceRepeatedly:                  FAIL
      Exception: Error: Unsupported feature: Type constructor: GHC.Prim.MutableByteArray#
      Context: Compiling type: GHC.Prim.MutableByteArray# GHC.Prim.RealWorld
      Context: Compiling data constructor type: GHC.ForeignPtr.MallocPtr
      Context: Compiling type: GHC.ForeignPtr.ForeignPtrContents
      Context: Compiling data constructor type: GHC.ForeignPtr.ForeignPtr
      Context: Compiling type: GHC.ForeignPtr.ForeignPtr GHC.Word.Word8
      Context: Compiling data constructor type: Data.ByteString.Internal.PS
      Context: Compiling type: Data.ByteString.Internal.ByteString
      Context: Compiling type: Data.ByteString.Internal.ByteString
                               -> PlutusTx.Builtins.Internal.BuiltinData
      Context: Compiling expr at "main:TH.Spec:(104,22)-(110,8)"
```

Reply:
Yeah, you have to do the other change I said too.

Reply:
I still get a failing test in `IsData.Spec`:

```
    bytestring:                       FAIL
      Exception: Error: Unsupported feature: Type constructor: GHC.Prim.MutableByteArray#
```

where `bytestring` test is:

```
    goldenUEval "bytestring" [plc (Proxy @"bytestring") (isDataRoundtrip (WrappedBS Builtins.emptyByteString))]
```

🤔

Reply:
It uses `Builtins.emptyByteString`:

```
{-# INLINABLE emptyByteString #-}
-- | An empty 'ByteString'.
emptyByteString :: BuiltinByteString
emptyByteString = BI.emptyByteString
```

where `BI.emptyByteString`:

```
{-# NOINLINE emptyByteString #-}
emptyByteString :: BuiltinByteString
emptyByteString = BuiltinByteString $ BS.empty
```

Is it because of this `BS.empty` definition?

Reply:
Hm, I tried `mustBeReplaced` and no, still fails.

Reply:
Maybe need to turn on debug-context to see what fails

Reply:
Hmm.. sorry but how to do it?

I can't anything useful in `tasty`, there is no proper verbosity options.

Reply:
Ah it's a plugin option. You can see at the top of the test files they have some plugin options in `OPTIONS_GHC` pragmas. You can change it to pass `debug-context`. Have a grep through the codebase and you'll see.

Reply:
Oh, this is very cool and helped. :)

Thanks!